census/keras/trainer/task.py

```diff
@@ -52,16 +53,20 @@ def __init__(self,
 
   def on_epoch_begin(self, epoch, logs={}):
     if epoch > 0 and epoch % self.eval_frequency == 0:
-      checkpoints = glob.glob(os.path.join(self.job_dir, 'checkpoint.*'))
-      checkpoints.sort()
-      census_model = load_model(checkpoints[-1])
-      census_model = model.compile_model(census_model, self.learning_rate)
-      loss, acc = census_model.evaluate_generator(
-          model.generator_input(self.eval_files, chunk_size=CHUNK_SIZE),
-          steps=self.steps)
-      print '\nEvaluation epoch[{}] metrics[{:.2f}, {:.2f}] {}'.format(
-          epoch, loss, acc, census_model.metrics_names)
 
+      checkpoints = glob.glob(path_join_for_non_gcs(self.job_dir, 'checkpoint.*'))
+      if len(checkpoints) > 0:
```

Comment:
This condition can potentially read a partially written file.

Reply:
Earlier I was checking % and epoch > 0 so I guess it would have skipped the first file. But in any case you want to process if len(checkpoints) > 1

Reply:
Sorry totally missed that. So len(checkpoints) still should be > 1 now that you have made it more bulletproof. Right?

Reply:
well, this is a safeguard against something not being in the filesystem, and will work fine as far as the remaining code goes. I like that it's independent of the epoch assertion and sanity checks the checkpoints.